### ISSUE TITLE
Vue syntax shorthand not working with hyphenated props

### ISSUE THREAD

**Issue by max-zinn** [johnsoncodehk/volar#4324]
# Description
We have the eslint rule 'vue/attribute-hyphenation' active in our project. However, this causes an error in my IDE (Webstorm 2024.1.2) when having camelCase props declared in a component and using shorthand when calling.

### Example

`ChildComponent.vue`
```vue
<script lang="ts" setup>
type Props = {
  camelCase: string
};
defineProps<Props>();
</script>
<template>
  <h1>{{camelCase}}</h1>
</template>
```

`ParentComponent.vue`
```vue
<script lang="ts" setup>
import { ref } from "vue";
import ChildComponent from "./ChildComponent.vue";

const camelCase = ref("Hey!");
</script>
<template>
<ChildComponent :camel-case /> <!-- <-- Error here -->
</template>
```

I searched for the issue on youtrack, because initially I thought it was a bug from webstorm and found [this thread](https://youtrack.jetbrains.com/issue/WEB-65974/Vue-3.4-v-bind-shorthand-case-hyphenation-issue#focus=Comments-27-9568983.0-0), where Jetbrains responded it should have something to do with the vue language server, but I couldn't find any issue here.

# Reproduction

You can review the issue in my repository I created for reproduction here: https://github.com/max-zinn/camelCasePropsTest

## Created new project with create-vue

```
npm create vue@latest
Need to install the following packages:
create-vue@3.10.3
Ok to proceed? (y) y

Vue.js - The Progressive JavaScript Framework

✔ Project name: … camelCasePropsTest
✔ Package name: … camelcasepropstest
✔ Add TypeScript? … No / **Yes**
✔ Add JSX Support? … **No** / Yes
✔ Add Vue Router for Single Page Application development? … **No** / Yes
✔ Add Pinia for state management? … **No** / Yes
✔ Add Vitest for Unit Testing? … **No** / Yes
✔ Add an End-to-End Testing Solution? › **No**
✔ Add ESLint for code quality? … No / **Yes**
✔ Add Prettier for code formatting? … No / **Yes**
✔ Add Vue DevTools 7 extension for debugging? (experimental) … **No** / Yes
```

## Installed versions

- vue@3.4.26
- typescript@5.4.5

## Error

It seems like a mathematical expression is recognized through the "-" causing `camel - case` to be a number, which is wrong. 

```
Vue: Property camel does not exist on type
{ $: ComponentInternalInstance; $data: {}; $props: Partial<{}> & Omit<{} & VNodeProps & AllowedComponentProps & ComponentCustomProps & Readonly<...>, never>;  ... 10 more ...; $watch<T extends string | ((...args: any) => any)>(source: T, cb: T extends (...args: any) => infer R ? (args_0: R, args_1: R) => any : (...ar...
Vue: Type number is not assignable to type string
```

# Reproduction link Vue SFC Playground

I also created a reproduction in Vue SFC online, where the error appears as well.

https://play.vuejs.org/#eNp9Uk1PwzAM/StRLt2k0YoPcRhlE6BJgwNMbMdIaGrdNaNNoiQtQ1X/O05Kx0Bjt9jv+fnZTkPvlArrCuiYxibRXFliwFZqwgQvldSWNERDRlqSaVmSAKnBHnqQpfrOh5ELnFJwwwQTiRTGEgvGzj9VDgJScuuEBozOoSgko0Mm4qhric0wsFCqYm0BI0Jirz12Amd5rxAhFEcHPDqi1mCrjG/CrZECp2hcNaMJlvMC9IuyHK0wOiYecdga+388+ZzVFYz6fJJD8n4kvzU7l2N0ocGAroHRPWbXegO2g2fLZ9jhew+WMq0KZJ8AX8HIonIeO9p9JVK0fcDzbh/9wrnYrMxsZ0GYfihn1DFbz2cUD+A299/oP3Yvwytfx0SLW+yPd+QbpJBxAQstlRl4ncOjjsnSarSFKkO8+6mDprz2D3zm55Om+f05plMSCElWmAvaNo6Q4quiruzv1d9q0G4H6BcHCS+uafsF5jvvaQ==

**Comment by so1ve** [comment]
Webstorm v2024.1 EAP doesn't use the latest vue language server, imo

**Comment by max-zinn** [comment]
@so1ve thanks for the hint!

I looked into Webstorms changelog and found [this article](https://blog.jetbrains.com/webstorm/2024/04/webstorm-2024-1/) where they stated they enabled Vue Language server by default now, but not which version. So I checked and found that my Webstorm is bundled with 1.8.27, but the latest version is 2.0.16, so I installed `@vue/language-server` globally via npm and selected it in webstorm. Now everything works fine, thank you! :) 

I'm considering requesting a feature to update it manually/automatically directly in WebStorm though. But I guess this issue can be closed as it's clear now that the issue didn't come from vue.
